clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -530,11 +531,6 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         // Invariant: non-explicit control flow (i.e., kernel-mediated) is indicated
         // by markers.  Checks are relaxed if a kernel event occurred so prev_instr_
         // is sufficient for now.
-        // XXX i#5912: For adding checks across every part of a signal handler we
-        // should change this to last_instr_in_cur_context_ and remove/adjust the
```

Comment:
But this doesn't seem to have been done: it's still passing prev_instr_ instead of last_instr_in_cur_context_.  Should we merge the two, for UNIX, to avoid confusion in the code?

Reply:
No, if I'm on the instr right after a sigreturn, then prev_instr will not equal last_*.  My suggestion as you saw was to merge them and not have them be two different things: do we ever actually want prev_instr?  (I'm not sure: you would have to quickly check the code.) 

It's been so long I don't remember what is already being checked and what this PR is adding.  For the instr after a sigreturn: have we already checked that transition from pre-interruption to interruption point?  Plus the resumption could be a repeat for a synchronous signal or auto-restart syscall: so it is special.  If that's the argument to do nothing then let's make that explicit with a comment.